allennlp/data/data_loaders/simple_data_loader.py

```diff
@@ -0,0 +1,91 @@
+import random
+from typing import Optional, List, Iterator
+
+from overrides import overrides
+import torch
+
+from allennlp.common.util import lazy_groups_of
+from allennlp.data.data_loaders.data_loader import DataLoader, allennlp_collate, TensorDict
+from allennlp.data.dataset_readers import DatasetReader
+from allennlp.data.instance import Instance
+from allennlp.data.vocabulary import Vocabulary
+import allennlp.nn.util as nn_util
+
+
+@DataLoader.register("simple", constructor="from_dataset_reader")
+class SimpleDataLoader(DataLoader):
+    """
+    A very simple `DataLoader` that is mostly used for testing.
+    """
+
+    def __init__(
+        self,
+        instances: List[Instance],
+        batch_size: int,
+        *,
+        shuffle: bool = False,
+        batches_per_epoch: Optional[int] = None,
+        vocab: Optional[Vocabulary] = None,
+    ) -> None:
+        self.instances = instances
+        self.batch_size = batch_size
+        self.shuffle = shuffle
+        self.batches_per_epoch = batches_per_epoch
+        self.vocab = vocab
+        self.cuda_device: Optional[torch.device] = None
+        self._batch_generator: Optional[Iterator[TensorDict]] = None
+
+    def __len__(self) -> int:
+        return len(list(lazy_groups_of(self.instances, self.batch_size)))
```

Comment:
Ouch. So inefficient. Why not `ceil(len(self.instances) / self.batch_size)`?

Reply:
That's better. I was being lazy 🙃